lib/pbench/server/auth/__init__.py

```diff
@@ -276,21 +413,27 @@ def token_introspect_online(self, token: str, token_info_uri: str) -> JSON:
                 "sub": <user_id>
             }
         """
+        if not self._tokeninfo_endpoint:
+            return None
+
         payload = {
             "client_id": self.client_id,
-            "client_secret": self.client_secret_key,
+            "client_secret": self._client_secret_key,
             "token": token,
         }
-        return self._post(token_info_uri, data=payload).json()
+        token_payload = self._connection.post(
+            self._tokeninfo_endpoint, data=payload
+        ).json()
```

Comment:
The successful response is a JSON and if we encounter an error during the connection or with our request then the error handling in `_method` function of the class `Connection` should be able to raise the `OpenIDClientError` which we handle in `auth.py` file in `verify_auth_oidc` line no 213.

Reply:
Thanks for making me look, @npalaska:  `post()` (or, actually, `_method()`) only returns normally in the success case; otherwise it raises an exception, and we'll never reach the point where we try to call `json()`.  So, this is fine.